test/test_train_imagenet.py

```diff
@@ -150,9 +165,17 @@ def train_loop_fn(model, loader, device, context):
             lr=FLAGS.lr,
             momentum=FLAGS.momentum,
             weight_decay=5e-4))
+
+    scheduler = get_scheduler_for_model(
+        FLAGS.model, optimizer, num_training_steps_per_epoch)
     tracker = xm.RateTracker()
     model.train()
     for x, (data, target) in loader:
+      # TODO BEFORE SUBMIT(zcain): Should we call scheduler.step() before or
```

Comment:
I think after, since why would you call before the first step for example? I checked fairseq also, they do it [after](https://github.com/pytorch/fairseq/blob/master/fairseq/trainer.py#L370-L371).

Reply:
I made changes to closer match the Pytorch lr_scheduler API, including the moving the scheduler.step() call after the optimizer.step() call. AFAICT from print statements, the correctness looks OK. Note that now I am calling super().init(), which itself makes a call to self.step():
https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L43